discovery_engine_core/tooling/bin/test_provider.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! A small binary utility to allow using the existing providers from a CLI.
+//!
+//! This can be useful for various debugging testing use cases.
+//!
+//! This calling this tool will fetch articles and then dump the `Article`
+//! **as returned by the provider** as json files.
+//!
+//! If you want raw input returned from the remote endpoint and feed into the
+//! provider use `curl` instead.
+//!
+//! [Here is the documentation for the remote newscatcher API.](https://docs.newscatcherapi.com/api-docs/endpoints/search-news)
+//!
+//! [And here for the remote gnews API.](https://gnews.io/docs/v4#search-endpoint)
+
+use std::{env, fs, path::PathBuf, thread::sleep, time::Duration};
+
+use anyhow::{bail, Context, Result};
+use tokio::runtime::Runtime;
+use url::Url;
+use xayn_discovery_engine_providers::{
+    gnews,
+    newscatcher,
+    Article,
+    CommonQueryParts,
+    Filter,
+    HeadlinesProvider,
+    HeadlinesQuery,
+    NewsProvider,
+    NewsQuery,
+};
+
+/// Some hard coded well known urls/paths which we likely currently use.
+mod well_known {
+    pub(super) const BASE_URL: &str = "https://api-gw.xaynet.dev";
```

Comment:
💅  maybe the base url could be also provided via env var and this would be a fallback value if the env var is not present, wdyt?

Reply:
Yes, through you can provide the whole url including domain if you want a different url, this is a hardly used testing tool, no point in making it flexible beyond the bare minimum.